pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1395,24 +1426,31 @@ func doInactivate(ctx *domainContext, status *types.DomainStatus, impatient bool
 		}
 	}
 
-	if err := hyper.Task(status).Delete(status.DomainName, status.DomainId); err != nil {
-		log.Errorf("Failed to delete domain %s (%v)", status.DomainName, err)
-	}
-	// Even if destroy failed we wait again
-	log.Functionf("doInactivate(%v) for %s: waiting for domain to be destroyed",
-		status.UUIDandVersion, status.DisplayName)
-
-	gone := waitForDomainGone(*status, maxDelay)
-	if gone {
-		status.DomainId = 0
+	if status.DomainId != 0 {
+		if err := hyper.Task(status).Delete(status.DomainName, status.DomainId); err != nil {
```

Comment:
Seems, that with moving this under condition, we will not fire `hyper.Task(status).Delete` in case of successful attempt of shutting down of xen domain ([first step](https://github.com/lf-edge/eve/blob/fc76ffe3f65c92d396ef9a157099531678af083e/pkg/pillar/hypervisor/xen.go#L487-L489), [second step](https://github.com/lf-edge/eve/blob/fc76ffe3f65c92d396ef9a157099531678af083e/pkg/pillar/cmd/domainmgr/domainmgr.go#L1771-L1779)), which one should stops the container with xen-start entrypoint. This container will never stops in this case (for kvm we rely on exit of the entrypoint process, i.e. qemu-system).

WDYT, @eriknordmark?

Reply:
Odd but I didn't see any failures in this space when I stress tested it.
@giggsoff can you easily reproduce a failure with Xen?

One question is if we do the Delete when DomainId is zero, should we also wait? The old code did that and it seemed to do it unnecessarily including when everything is gone. So we need to make sure we don't wait for 10 extra minutes for no reason.